### PR TITLE
CORE-749 Add fetchTag helper function

### PR DESCRIPTION
**Pull request by ghmeier**
So we can dynamically resolve environment-names from the instances
themselves, adds the fetchTag helper function that retrieves ec2 tags
from the instance based on its identity document.

Also adds tests, eslint, and prettier.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -4,20 +4,76 @@
...
+ * @param      {string}   tag     The tag
+ * @return     {Promise<string>}  The value of the tag, or nul if it's not found.
+ */
+async function fetchTag(tag) {
+  const { instanceId, region } = await fetchInstanceIdentity();
+  const ec2 = new AWS.EC2({
+    region,
+  });
+
+  const result = await ec2
```
Comment (ttacon) [inline index.js]: Will instances need extra permission in order to run this?

Comment (ghmeier) [reply index.js]: `custom-cloudwatch-metrics` allows `ec2:DescribeTags` and is part of our `aws-elasticbeanstalk-ec2-role`:
image

File: index.js [hunk 2]
```diff
@@ -4,20 +4,76 @@
...
- * instead.
+ * fetch fetches the given metadata field for the currently running instance.
  * @param  {String}   field The metadata field to retrieve.
  * @param  {Promise<String>} A promise that resolves to the field data.
  */
 function fetch(field) {
-  // We use the IP address  as it is referenced from the AWS docs:
-  // https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html
-  return got(`http://169.254.169.254/latest/meta-data/${field}`).then((res) => res.text);
+  return got(`${baseUrl}/meta-data/${field}`).then((res) => res.text);
```
Comment (skeggse) [inline index.js]: Could we use the [AWS `MetadataService`](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/MetadataService.html) instead?

File: package.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
   "description": "A utility for retrieving the desired AWS EC2 instance metadata from a running instance",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1",
+    "test": "jest",
```
Comment (skeggse) [inline package.json]: :tada:

**Pull request merged**